src/devices/Ssd1351/Ssd1351Commands.cs

```diff
@@ -0,0 +1,541 @@
+        /// <param name="endRow">Row end address with a range of 0-127.</param>
+        public void SetRowAddress(byte startRow = 0x00, byte endRow = 0x7F)
+        {
+            if (startRow > 0x7F)
+            {
+                throw new ArgumentException("The row start address is invalid.", nameof(startRow));
+            }
+
+            if (endRow > 0x7F)
+            {
+                throw new ArgumentException("The row end address is invalid.", nameof(endRow));
+            }
+
+            SendCommand(Ssd1351Command.SetRow, startRow, endRow);
+        }
+
+        /// <summary>
+        /// This command sets the divide ratio to generate DCLK (Display Clock) from CLK and
+        /// programs the oscillator frequency Fosc that is the source of CLK if CLS pin is pulled high.
+        /// </summary>
+        /// <param name="displayClockDivideRatio">Display clock divide ratio with a range of 0-15.</param>
+        /// <param name="oscillatorFrequency">Oscillator frequency with a range of 0-15.</param>
+        public void SetDisplayClockDivideRatioOscillatorFrequency(byte displayClockDivideRatio = 0x01, byte oscillatorFrequency = 0x0D)
+        {
+            if (displayClockDivideRatio > 0x0F)
+            {
+                throw new ArgumentException("The display clock divide ratio is invalid.", nameof(displayClockDivideRatio));
+            }
+
+            if (oscillatorFrequency > 0x0F)
+            {
+                throw new ArgumentException("The oscillator frequency is invalid.", nameof(oscillatorFrequency));
+            }
+
+            SendCommand(Ssd1351Command.SetClockDiv, (byte)((oscillatorFrequency << 4) | displayClockDivideRatio));
+        }
+
+        /// <summary>
+        /// This command is used to set Contrast Setting of the display.
+        /// The chip has 256 contrast steps from 00h to FFh.  The segment
+        /// output current ISEG increases linearly with the contrast step,
+        /// which results in brighter display.
+        /// </summary>
+        /// <param name="colorAContrast">Contrast level for color A.</param>
+        /// <param name="colorBContrast">Contrast level for color B.</param>
+        /// <param name="colorCContrast">Contrast level for color C.</param>
+        public void SetContrastABC(byte colorAContrast = 0x86, byte colorBContrast = 0x51, byte colorCContrast = 0x86)
+        {
+            SendCommand(Ssd1351Command.SetContrastABC, colorAContrast, colorBContrast, colorCContrast);
+        }
+
+        /// <summary>
+        /// This command specifies the mapping of the display start line to one of COM0-COM127
+        /// (assuming that COM0 is the display start line then the display start line register is equal to 0).
+        /// </summary>
+        /// <param name="displayOffset">Display offset with a range of 0-127.</param>
+        public void SetDisplayOffset(byte displayOffset = 0x60)
+        {
+            if (displayOffset > 0x7F)
```

Comment:
lower bound check. If you want to keep it as you have it and still check lower bound for all of these, you can just cast the value you are checking to a uint. this will work as long as the lower bound is 0, because all negative numbers will overflow and become very large numbers, all larger than your upper bounds so you only need one check.

Reply:
@joperezr 
Am I misunderstanding something... these parameters are bytes and hence cannot go below 0 and so I have not checked for lower bounds. I believe I do need to check that start values are <= end values though.

Reply:
you are right, that was my bad I didn't see that all these were bytes. I originaly thought they were ints. You can disregard all related comments.